From ebad70945cd52a8450c234a947c64f1a16ff62ea Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 28 Nov 2022 19:07:07 +0200
Subject: [PATCH] [SW-0] module_lkd: fix msg validation

The message validation should not include the version, as this will
impose compatibility problem with future versions.
Omit the version from validation.

Change-Id: I5487adeb92f4bccab88f7f2526c23212296d9f4c
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index 24b4ef8..94838ff 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -409,8 +409,7 @@
 	struct lkd_fw_comms_msg *mcomms = (struct lkd_fw_comms_msg *)offset;
 	int err = 1;
 
-	if (HL_COMMS_MSG_MAGIC_VALIDATE(mcomms->header.magic,
-					HL_COMMS_MSG_MAGIC_V1)) {
+	if (HL_COMMS_MSG_MAGIC_VALIDATE_MAGIC(mcomms->header.magic)) {
 		sts.exec_res = FOUND_FAST_RETURN;
 		err = 0;
 	}
